interview-questions.md

```diff
@@ -8,47 +8,55 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+  Researched answer: A function can have parameters that are in place of arguments. However, higher order functions have parameters that are predifined. Function arguments are values and if omitted while invoking a function, the output will be undefined.
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The method .map() takes value, index, and array as its predefined parameters. Value is a required parameter while index and array are optional. Map can extract the index of an element and return the value along with the changes made to the value of an array.
 
-  Researched answer:
+  Researched answer: The higher order functions .map has three predefined parameters. These parameters must be in order of value, index, then array. Although we cannot change the order of the paramters, we are able to rename them. It is common practice when using more than one parameter to label the outputs of an array to indicate whether what is being logged is the value or index of the elements in the array. 
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: The higher order function map will iterate over an array and return a new array of the same length. Filter on the other hand iterates over an array but returns an array of shorter length or elements removed.
 
-  Researched answer:
+  Researched answer: map and filter act only on arrays. Map modifies each element in an array and returns these values in a new array. Filter removes unwanted elements based on a set specifications and returns a new array reflecting the changes.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: Functions are reusable code that contain logic. They must be invoked to initate its logic/behavior. Methods are also functions however, methods belong to objects and are used to add logic to objects.
 
-  Researched answer:
+  Researched answer: Functions and methods perform an action based on inputs or specifications. The difference is that methods give functionality to objects. this is a method  that uses dot notation and is used to reference an object inside of itself. 
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructing provides a pathway to nested key values within a object. Instead of logging the entire pathway to output an objects value, with object destructuring we can assign a variable to the pathway of the nested key.
 
-  Researched answer:
+  Researched answer: Object destructring can make your code clearer and provides a more effiecient way of referencing values in an object. When creating a pathway to a value in an object and assiging it a variable we use curly braces. Within the curly brace is the key and is assigned the value of the pathway.
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: 
```

Comment:
Awesome job!  Again, you seem to have a good understanding of vocabulary and concepts.  Love that you are using examples.  Excellent!